File: python/ex008.py
# leia um valor em metro e a exiba convertido em centimetros e milimetros.
while True:
    entry = input("Uma distância em metros: ")
    try:
        distance = float(entry)
        break
    except ValueError:
        print("Por favor, digite um número válido.")

print("A média de {} corresponse a \n{:.3f}km\n{:.2f}hm\n{:.1f}dam\n{}dm\n{}cm\n{}mm".format(
    distance,
    distance / 1000, # Kilometros
    distance / 100, # Hectometros
    distance / 10, # Decametros
    distance * 10, # Decimetros
    distance * 100, # Centimetros
    distance * 1000 # Milimetros
))

File: rust/ex008.rs
use std::io;

fn main() {
    let distance = read_number("Uma distância em metros: ");
    println!(
        "A média de {} corresponde a \n{:.3}km\n{:.2}hm\n{:.1}dam\n{}dm\n{}cm\n{}mm",
        distance, distance / 1000.0, distance / 100.0, distance / 10.0, distance * 10.0, distance * 100.0, distance * 1000.0
    );
}

fn read_number(prompt: &str) -> f32 {
    loop {
        let mut number = String::new();
        println!("{prompt}");

        io::stdin()
            .read_line(&mut number)
            .expect("Erro ao ler input");

        match number.trim().parse::<f32>() {
            Ok(num) => return num,
            Err(_) => {
                println!("Por favor, digite um numero valido.");
            }
        }
    }
}
